dbms/src/Storages/Page/V3/WAL/WALReader.cpp

```diff
@@ -242,14 +248,14 @@ std::optional<String> WALStoreReader::next()
         std::tie(ok, record) = reader->readRecord();
         if (ok)
         {
-            return record;
+            return std::make_pair(reading_checkpoint_file, record);
```

Comment:
Can we use `!checkpoint_read_done` instead of `reading_checkpoint_file` here?

Reply:
`checkpoint_read_done`  is actually mean whether we need to create a reader from checkpoint file. After create the reader, it will be set to true first before read it.

Reply:
If we want to eliminate the member `reading_checkpoint_file`, we can make it a return value of `openNextFile` instead.

Reply:
OK